yellowbrick/classifier/prcurve.py

```diff
@@ -0,0 +1,472 @@
+            for f1 in np.linspace(0.2, 0.8, num=4):
+                x = np.linspace(0.01, 1)
+                y = f1 * x / (2 * x - f1)
+                self.ax.plot(x[y>=0], y[y>=0], color='#333333', alpha=0.2)
+                self.ax.annotate('$f_1={:0.1f}$'.format(f1), xy=(0.9, y[45]+0.02))
+
+        if self.target_type_ == BINARY:
+            return self._draw_binary()
+        return self._draw_multiclass()
+
+    def _draw_binary(self):
+        """
+        Draw the precision-recall curves in the binary case
+        """
+        self._draw_pr_curve(self.recall_, self.precision_, label="binary PR curve")
+        self._draw_ap_score(self.score_)
+
+
+    def _draw_multiclass(self):
+        """
+        Draw the precision-recall curves in the multiclass case
+        """
+        # TODO: handle colors better with a mapping and user input
+        if self.per_class:
+            for cls in self.classes_:
+                precision = self.precision_[cls]
+                recall = self.recall_[cls]
+
+                label = "PR for class {} (area={:0.2f})".format(cls, self.score_[cls])
+                self._draw_pr_curve(recall, precision, label=label)
+
+        if self.micro:
+            precision = self.precision_[MICRO]
+            recall = self.recall_[MICRO]
+            self._draw_pr_curve(recall, precision)
+
+        self._draw_ap_score(self.score_[MICRO])
+
+    def _draw_pr_curve(self, recall, precision, label=None):
+        """
+        Helper function to draw a precision-recall curve with specified settings
+        """
+        self.ax.step(
+            recall, precision, alpha=self.line_opacity, where='post', label=label
+        )
+        if self.fill_area:
+            self.ax.fill_between(recall, precision, step='post', alpha=self.fill_opacity)
+
+    def _draw_ap_score(self, score, label=None):
+        """
+        Helper function to draw the AP score annotation
+        """
+        label = label or "AP={:0.2f}".format(score)
+        if self.ap_score:
+            self.ax.axhline(
+                y=score, color="r", ls="--", label=label
+            )
+
+    def finalize(self):
+        """
+        Finalize the figure by adding titles, labels, and limits.
+        """
+        self.set_title("Precision-Recall Curve for {}".format(self.name))
+        self.ax.legend(loc='lower left', frameon=True)
+
+        self.ax.set_xlim([0.0, 1.0])
+        self.ax.set_ylim([0.0, 1.0])
+
+        self.ax.set_ylabel("Precision")
+        self.ax.set_xlabel("Recall")
+
+    def _get_y_scores(self, X):
+        """
+        The ``precision_recall_curve`` metric requires target scores that
+        can either be the probability estimates of the positive class,
+        confidence values, or non-thresholded measures of decisions (as
+        returned by a "decision function").
+
+        .. todo:: refactor shared method with ROCAUC
+        .. todo:: ensure only probabilities from positive case are returned (binary)
```

Comment:
Great catch @zjpoh! Let's use the  `..todo::` style instead of `#TODO` inside docstrings, since Sphinx will skip over any `..todo::`'s when building the docs, which looks a bit nicer.

Reply:
@rebeccabilbro I agree - this should be a docstring vs. non-docstring thing; so basically I was completely wrong; in the docstring above (as you pointed out), that should become `.. todo::` since it's a documentation todo, and these should probably come out of the docstring and become single line `# TODO` comments in the code!

Reply:
And added to an issue ... which I'll also do.